raiden/messages/metadata.py

```diff
@@ -24,20 +24,9 @@ def __init__(
         self._validate_address_metadata()
 
     def _validate_address_metadata(self) -> None:
-        if self.address_metadata is None:
-            return
-
-        for address, metadata in list(self.address_metadata.items()):
-            user_id = metadata.get("user_id")
-            displayname = metadata.get("displayname")
-
-            if user_id is None or displayname is None:
-                del self.address_metadata[address]  # we can't verify this user's identity
-                continue
-
-            verified_address = validate_user_id_signature(user_id, displayname)  # type: ignore
-            if verified_address != address:
-                del self.address_metadata[address]
+        validation_errors = validate_address_metadata(self)
+        for address in validation_errors:
+            del self.address_metadata[address]
```

Comment:
Changing the address metadata will change the message hash and thus break the signature check if it is done this early.

Reply:
Yeah, we found this while I was testing with the LC: https://github.com/raiden-network/raiden/issues/7052#issuecomment-843202262

Reply:
I think so, but I have also not tested it. I just expect it to break in the same way as https://github.com/raiden-network/raiden/issues/7052#issuecomment-843198423